eureka2-client/src/main/java/com/netflix/eureka2/client/registration/RegistrationTracker.java

```diff
@@ -0,0 +1,107 @@
+package com.netflix.eureka2.client.registration;
+
+import com.netflix.eureka2.registry.instance.InstanceInfo;
+import com.netflix.eureka2.utils.rx.NoOpSubscriber;
+import rx.Observable;
+import rx.Subscriber;
+import rx.subjects.BehaviorSubject;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * TODO deprecate me
+ *
+ * This is a bridging class that bridges the registration behaviour of the current eurekaClient (individual register()
+ * and unregister() calls) to the new behaviour in the handler (a stream of instanceInfos). Going forward, this class
+ * should go away as we change over the client api.
+ *
+ * @author David Liu
+ */
+public class RegistrationTracker {
```

Comment:
In the current model we never onError to the client. It is fine, assuming we will refactor client API, and expose different result type.